benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp

```diff
@@ -234,6 +234,56 @@ void GraphicsBenchmarkApp::Setup()
     }
 }
 
+void GraphicsBenchmarkApp::SetupMetrics()
+{
+    Application::SetupMetrics();
+    ppx::metrics::MetricMetadata metadata                     = {ppx::metrics::MetricType::GAUGE, "CPU Submission Time", "ms", ppx::metrics::MetricInterpretation::LOWER_IS_BETTER, {0.f, 10000.f}};
+    mMetricsData.metrics[MetricsData::kTypeCPUSubmissionTime] = AddMetric(metadata);
+    PPX_ASSERT_MSG(mMetricsData.metrics[MetricsData::kTypeCPUSubmissionTime] != ppx::metrics::kInvalidMetricID, "Failed to add CPU Submission Time metric");
+
+    metadata                                          = {ppx::metrics::MetricType::GAUGE, "Bandwidth", "GB/s", ppx::metrics::MetricInterpretation::HIGHER_IS_BETTER, {0.f, 10000.f}};
+    mMetricsData.metrics[MetricsData::kTypeBandwidth] = AddMetric(metadata);
+    PPX_ASSERT_MSG(mMetricsData.metrics[MetricsData::kTypeBandwidth] != ppx::metrics::kInvalidMetricID, "Failed to add Bandwidth metric");
+}
+
+void GraphicsBenchmarkApp::UpdateMetrics()
+{
+    uint64_t frequency = 0;
+    GetGraphicsQueue()->GetTimestampFrequency(&frequency);
+
+    ppx::metrics::MetricData data = {ppx::metrics::MetricType::GAUGE};
+    data.gauge.seconds            = GetElapsedSeconds();
+    data.gauge.value              = mCPUSubmissionTime;
+    RecordMetricData(mMetricsData.metrics[MetricsData::kTypeCPUSubmissionTime], data);
+
+    const float    gpuWorkDurationInSec = static_cast<float>(mGpuWorkDuration / static_cast<double>(frequency));
+    const uint32_t width                = GetSwapchain()->GetWidth();
+    const uint32_t height               = GetSwapchain()->GetHeight();
+    const uint32_t quadCount            = pFullscreenQuadsCount->GetValue();
+
+    if (quadCount) {
+        // Skip the first kSkipFrameCount frames after the knob of quad count being changed to avoid noise
+        constexpr uint32_t kSkipFrameCount  = 2;
+        static uint32_t    skipFrameCounter = 0;
+        if (pFullscreenQuadsCount->DigestUpdate()) {
+            skipFrameCounter = kSkipFrameCount;
+        }
+
+        if (skipFrameCounter == 0) {
+            const float dataWriteInGb = (static_cast<float>(width) * static_cast<float>(height) * 4.f * quadCount) / (1024.f * 1024.f * 1024.f);
```

Comment:
this is not tracked by the metrics system since it depends on the number of quads, if the user changes the quad count, the average/min/max do not make much sense anymore.

On the other hand, i still want to show this value in the UI so that we know what is the total amount of data being output.

I could cache this value, but since it is not really expensive to calculate this, it is better to just leave it calculated two times.